rules_java_gapic/java_gapic_spring.bzl

```diff
@@ -0,0 +1,141 @@
+    gapic_srcjar = ctx.file.gapic_srcjar
+    output_srcjar_name = ctx.label.name
+    srcjar_name = output_srcjar_name + "_raw.srcjar"
+    output_spring = ctx.outputs.spring
+    formatter = ctx.executable.formatter
+
+    output_dir_name = ctx.label.name
+    output_dir_path = "%s/%s" % (output_spring.dirname, output_dir_name)
+
+    script = """
+    WORKING_DIR=`pwd`
+    cd $WORKING_DIR
+    mkdir {output_dir_path}/spring
+    unzip -q {gapic_srcjar}
+    unzip -q temp-codegen-spring.srcjar -d {output_dir_path}
+    # This may fail if there are spaces and/or too many files (exceed max length of command length).
+    {formatter} --replace $(find {output_dir_path}/spring -type f -printf "%p ")
+
+    # Spring source files.
+    cd {output_dir_path}/src/main/java
+    zip -r $WORKING_DIR/{output_srcjar_name}.srcjar ./
+
+    cd $WORKING_DIR
+
+    mv $WORKING_DIR/{output_srcjar_name}.srcjar {output_spring}
+    """.format(
+        output_srcjar_name = output_srcjar_name,
+        gapic_srcjar = gapic_srcjar.path,
+        srcjar_name = srcjar_name,
+        formatter = formatter,
+        output_dir_name = output_dir_name,
+        output_dir_path = output_dir_path,
+        output_spring = output_spring.path,
+    )
+
+    ctx.actions.run_shell(
+        inputs = [gapic_srcjar],
+        tools = [formatter],
+        command = script,
+        outputs = [output_spring],
+    )
+
+_java_gapic_spring_postprocess_srcjar = rule(
+    attrs = {
+        "gapic_srcjar": attr.label(mandatory = True, allow_single_file = True),
+        "formatter": attr.label(
+            default = Label("//:google_java_format_binary"),
+            executable = True,
+            cfg = "host",
+        ),
+    },
+    outputs = {
+        "spring": "%{name}-spring.srcjar",
+    },
+    implementation = _java_gapic_spring_postprocess_srcjar_impl,
+)
+
+def java_gapic_spring_library(
+        name,
+        srcs,
+        grpc_service_config = None,
+        gapic_yaml = None,
+        service_yaml = None,
+        transport = None,
+        **kwargs):
+    library_name = name + "-spring"
+    raw_srcjar_name = name + "_raw"
+
+    _java_gapic_spring_srcjar(
+        name = raw_srcjar_name,
+        srcs = srcs,
+        grpc_service_config = grpc_service_config,
+        gapic_yaml = gapic_yaml,
+        service_yaml = service_yaml,
+        transport = transport,
+        **kwargs
+    )
+
+    _java_gapic_spring_postprocess_srcjar(
+        name = name,
+        gapic_srcjar = "%s.srcjar" % raw_srcjar_name,
+        **kwargs
+    )
+
+def _java_gapic_spring_srcjar(
+        name,
+        srcs,
+        grpc_service_config,
+        gapic_yaml,
+        service_yaml,
+        transport,
+        # Can be used to provide a java_library with a customized generator,
+        # like the one which dumps descriptor to a file for future debugging.
+        java_generator_name = "java_gapic_spring",
+        **kwargs):
+    file_args_dict = {}
+
```

Comment:
ah-ha! I think this might be the missing piece:
`grpc_service_config` is never processed/passed-in
`java_gapic.bzl` ref: https://github.com/googleapis/gapic-generator-java/blob/abb931d7dc5fde126aec13eea395c4ad966e1b1d/rules_java_gapic/java_gapic.bzl#L199-L200